### ISSUE TITLE
Console Crash when Dragging Items

### ISSUE THREAD

**Issue by hugovdev** [Minestom/Minestom#567]
When dragging items on an inventory I get this crash:
`java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:699)
	at java.base/java.nio.DirectByteBuffer.get(DirectByteBuffer.java:324)
	at net.minestom.server.utils.binary.BinaryReader.readByte(BinaryReader.java:59)
	at net.minestom.server.utils.binary.BinaryReader.read(BinaryReader.java:259)
	at java.base/java.io.DataInputStream.readByte(DataInputStream.java:271)
	at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readFullyFormedTag(Extensions.kt:12)
	at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion$readContents$$inlined$Kompound$1.run(NBT.kt:222)
	at org.jglrxavpok.hephaistos.nbt.NBT$Companion.Compound(NBT.kt:108)
	at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:96)
	at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:58)
	at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readTag(Extensions.kt:27)
	at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readFullyFormedTag(Extensions.kt:18)
	at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion$readContents$$inlined$Kompound$1.run(NBT.kt:222)
	at org.jglrxavpok.hephaistos.nbt.NBT$Companion.Compound(NBT.kt:108)
	at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:96)
	at org.jglrxavpok.hephaistos.nbt.NBTCompound$Companion.readContents(NBTCompound.kt:58)
	at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readTag(Extensions.kt:27)
	at org.jglrxavpok.hephaistos.nbt.ExtensionsKt.readFullyFormedTag(Extensions.kt:18)
	at org.jglrxavpok.hephaistos.nbt.NBTReader.readNamed(NBTReader.kt:48)
	at org.jglrxavpok.hephaistos.nbt.NBTReader.read(NBTReader.kt:58)
	at net.minestom.server.utils.binary.BinaryReader.readTag(BinaryReader.java:274)
	at net.minestom.server.utils.NBTUtils.readItemStack(NBTUtils.java:116)
	at net.minestom.server.utils.binary.BinaryReader.readItemStack(BinaryReader.java:193)
	at net.minestom.server.network.packet.client.play.ClientClickWindowPacket$ChangedSlot.<init>(ClientClickWindowPacket.java:39)
	at net.minestom.server.utils.binary.BinaryReader.readList(BinaryReader.java:248)
	at net.minestom.server.utils.binary.BinaryReader.readVarIntList(BinaryReader.java:238)
	at net.minestom.server.network.packet.client.play.ClientClickWindowPacket.<init>(ClientClickWindowPacket.java:23)
	at net.minestom.server.network.packet.client.ClientPacketsHandler.create(ClientPacketsHandler.java:35)
	at net.minestom.server.network.PacketProcessor.create(PacketProcessor.java:33)
	at net.minestom.server.network.PacketProcessor.process(PacketProcessor.java:48)
	at net.minestom.server.network.player.PlayerSocketConnection.processPackets(PlayerSocketConnection.java:109)
	at net.minestom.server.network.socket.Worker.lambda$run$0(Worker.java:74)
	at java.base/sun.nio.ch.SelectorImpl.processReadyEvents(SelectorImpl.java:294)
	at java.base/sun.nio.ch.WEPollSelectorImpl.processEvents(WEPollSelectorImpl.java:180)
	at java.base/sun.nio.ch.WEPollSelectorImpl.doSelect(WEPollSelectorImpl.java:116)
	at java.base/sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:129)
	at java.base/sun.nio.ch.SelectorImpl.select(SelectorImpl.java:167)
	at net.minestom.server.network.socket.Worker.run(Worker.java:63)
`

Worth pointing out that I have a inventorycondition set up for the player's inventory that cancels every move.

**Comment by TheMode** [comment]
I will need a proper reproducer.

Have you been able to reproduce in the 1.18 branch?

**Comment by hugovdev** [comment]
> 

Yeah, 1.18 branch.
If I'm not mistaken this is how you reproduce it.

1- Create an inventory with items inside.
2- Add an inventorycondition to the player inventory that setCancel to true.
3- Open the inventory and drag an item to another slot in that same inventory.

I don't know if this error is cause by the inventory condition or not, but this is my current setup.

**Comment by TheMode** [comment]
By reproducer I meant code, you should be able to make a single-class server and send it here.
In the current state, I have obviously no way to reproduce your setup (could be related to the item nbt, inventory, inventory condition, etc) so we are playing the guessing game.

**Comment by hugovdev** [comment]
`package me.hugo.thankmaslobby;

import me.hugo.thankmaslobby.item.HotBarItem;
import me.hugo.thankmaslobby.util.SkinUtil;
import me.hugo.thankmaslobby.world.EmptyGenerator;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextDecoration;
import net.minestom.server.MinecraftServer;
import net.minestom.server.coordinate.Pos;
import net.minestom.server.entity.Player;
import net.minestom.server.event.GlobalEventHandler;
import net.minestom.server.event.player.PlayerBlockInteractEvent;
import net.minestom.server.event.player.PlayerLoginEvent;
import net.minestom.server.extras.MojangAuth;
import net.minestom.server.instance.Instance;
import net.minestom.server.instance.InstanceContainer;
import net.minestom.server.instance.InstanceManager;
import net.minestom.server.inventory.Inventory;
import net.minestom.server.inventory.InventoryType;
import net.minestom.server.item.ItemStack;
import net.minestom.server.item.Material;
import net.minestom.server.item.metadata.PlayerHeadMeta;
import net.minestom.server.utils.NamespaceID;
import net.minestom.server.world.DimensionType;

import java.util.UUID;

public class Main {

    public static void main(String[] args) {
        MinecraftServer minecraftServer = MinecraftServer.init();
        MojangAuth.init();
        GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();

        Inventory testInventory = new Inventory(InventoryType.CHEST_5_ROW, "Test Inventory");
        testInventory.addItemStack(ItemStack.builder(Material.PLAYER_HEAD)
                .meta(PlayerHeadMeta.class, meta -> meta.playerSkin(SkinUtil.LOCKED_STATE).skullOwner(UUID.randomUUID())
                        .displayName(Component.text("???", NamedTextColor.RED).decoration(TextDecoration.BOLD,true).decoration(TextDecoration.ITALIC, false))
                        .lore(Component.text("Skin Secret", NamedTextColor.DARK_GRAY).decoration(TextDecoration.ITALIC, false), Component.text(""),
                                Component.text("This secret is locked!", NamedTextColor.GRAY).decoration(TextDecoration.ITALIC, false), Component.text(""),
                                Component.text("Locked!", NamedTextColor.RED).decoration(TextDecoration.ITALIC, false))
                        .build()).build());
        initMainWorld();

        globalEventHandler.addListener(PlayerLoginEvent.class, event -> {
            final Player player = event.getPlayer();
            // Set player spawn location
            event.setSpawningInstance(MinecraftServer.getInstanceManager().getInstances().iterator().next());
            player.setRespawnPoint(new Pos(0, 40, 0));

            player.getInventory().addInventoryCondition((player1, i, clickType, inventoryConditionResult) -> {
                inventoryConditionResult.setCancel(true);
            });
        });

        globalEventHandler.addListener(PlayerBlockInteractEvent.class, playerBlockInteractEvent -> {
            playerBlockInteractEvent.getPlayer().openInventory(testInventory);
        });

        //ThankmasLobby.main(args);

        minecraftServer.start("0.0.0.0", 25565);
    }

    public static final NamespaceID DIMENSION_ID = NamespaceID.from("thankmas:lobby_world");
    public static final DimensionType DIMENSION_TYPE = DimensionType.builder(DIMENSION_ID).ambientLight(1.0f).build();

    public static Instance initMainWorld() {
        if (!MinecraftServer.getDimensionTypeManager().isRegistered(DIMENSION_ID))
            MinecraftServer.getDimensionTypeManager().addDimension(DIMENSION_TYPE);

        InstanceManager instanceManager = MinecraftServer.getInstanceManager();
        InstanceContainer instanceContainer = instanceManager.createInstanceContainer(DIMENSION_TYPE);
        instanceContainer.setChunkGenerator(new EmptyGenerator());

        MinecraftServer.getInstanceManager().registerInstance(instanceContainer);
        instanceContainer.loadChunk(0, 0).join();

        return instanceContainer;
    }
}
`

Here ya go, tested it and it happens. Apparently it does have to do with NBTs, I tried with a normal named item and it didn't happen.

**Comment by hugovdev** [comment]
Can't get it to format it right sorry xD

**Comment by BomBardyGamer** [comment]
Use three backticks (\`\`\`) rather than just the one

**Comment by hugovdev** [comment]
> Use three backticks (```) rather than just the one

ty

**Comment by iam4722202468** [comment]
I'm having the same issue. I think it's a packet compression problem, as it only happens with larger item NBT packets and disabling compression solves the problem.

Here's a way to reproduce the issue

```.addListener(PlayerSpawnEvent.class, event -> {
                final Player player = event.getPlayer();
                player.setGameMode(GameMode.SURVIVAL);
                player.setPermissionLevel(2);
                {
                    Component c = Component.text("adsf").append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf")).append(Component.text("adsf"));
                    ItemStack i = ItemStack.builder(Material.STONE).amount(50).meta(itemMetaBuilder -> itemMetaBuilder.displayName(c)).build();
                    player.getInventory().addItemStack(i);
                }
            });```

Dragging the stack will reproduce the error.

**Comment by TheMode** [comment]
Fixed by 87d5a33c760227c18a75ef86ac5002c9136a8817
